spikeinterface/core/core_tools.py

```diff
@@ -703,6 +704,10 @@ def _write_zarr_chunk(segment_index, start_frame, end_frame, worker_ctx):
     traces = traces.astype(dtype)
     zarr_dataset[start_frame:end_frame, :] = traces
 
+    # fix memory leak by forcing garbage collection
+    del traces
+    gc.collect()
```

Comment:
does it works without the gc.collect() ?

Reply:
No 😧

Reply:
I mean it works but the jobs keep accumulating RAM

Reply:
This is also something that I think we encountered with hdf5 or memap arrays. You need to regularly force the destruction of the objects to clear the built-in caches that they have, in order to prevent memory leaks